### PR TITLE
Badge mode

### PR DESCRIPTION
**Pull request by lemieux**
This bring badge mode support for the widget.

Since the animation is using scaling, the input was getting 0px as its width on initial render. I fixed that by using flexboxes instead (which I tested on Chrome, iOS Safari, Chrome Android, and IE10 so far). That also fixes the behavior we saw in embedded mode and removes the need of doing it in JS.

@chloepouprom @dannytranlx @jugarrit @mspensieri @alavers

### CODE REVIEW

File: src/stylesheets/animations.less [hunk 1]
```diff
@@ -1,209 +1,207 @@
...
     }
-}
 
-.sk-close {
-    bottom: @widget-close-bottom-md;
+    &.sk-close {
+        bottom: @widget-close-bottom-md;
 
-    -webkit-animation: sk-close-frames-md .4s cubic-bezier(.62, .28, .23, .99);
             animation: sk-close-frames-md .4s cubic-bezier(.62, .28, .23, .99);
```
Comment (chloepouprom) [inline src/stylesheets/animations.less]: Are all the `sk-close-frames` related to the tab widget? If so, I would rename to `sk-close-tab-frames` to be consistent with the `sk-close-badge-frames`

Comment (lemieux) [inline src/stylesheets/animations.less]: good point

**Comment by lemieux** [comment]
Small comment : I removed all the -webkit prefix from animations because it seems they are useless now.

File: src/js/components/connect-notification.jsx [hunk 2]
```diff
@@ -66,9 +66,9 @@ export class ConnectNotificationComponent extends Component {
                     const separator = index !== array.length - 1 ? ',' : '';
 
                     return <div style={ linkStyle }
-                                className='channel-details'>
-                               <a key={ channel.type }
-                                  style={ linkStyle }
+                                className='channel-details'
+                                key={channel.type}>
```
Comment (chloepouprom) [inline src/js/components/connect-notification.jsx]: Why did you move the key from `<a>` to `<div>`?

Comment (lemieux) [inline src/js/components/connect-notification.jsx]: because the key needs to be at the top level element of an item in a list.

Comment (lemieux) [inline src/js/components/connect-notification.jsx]: it was throwing warning about it

**Comment by chloepouprom** [comment]
👍

File: src/js/components/badge.jsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import React, { Component, PropTypes } from 'react';
+
+import { openWidget } from '../services/app-service';
+import { SK_DARK_CONTRAST } from '../constants/styles';
+
+export class DefaultIcon extends Component {
```
Comment (Mario54) [inline src/js/components/badge.jsx]: nit: rename `DefaultBadgeIcon`?

Comment (lemieux) [inline src/js/components/badge.jsx]: sure, but your're already in the "badge" module /shrug

File: src/js/components/widget.jsx [hunk 4]
```diff
@@ -105,39 +109,46 @@ export class WidgetComponent extends Component {
...
-                    { channelsComponent }
-                    <Conversation />
-                    { footer }
-                </div>
-            </div>
-            );
+        let badge;
+
+        if (displayStyle === DISPLAY_STYLE.BADGE && !appState.embedded) {
+            badge = <Badge shown={ !appState.widgetOpened } />;
```
Comment (Mario54) [inline src/js/components/widget.jsx]: Why is it passing a `shown` prop instead of setting `badge` to `null` if widgetOpened is true?

File: src/js/constants/styles.js [hunk 5]
```diff
@@ -3,5 +3,9 @@ export const ANIMATION_TIMINGS = {
 };
 
 export const SK_PURPLE = '#92278f';
+export const SK_DARK_CONTRAST = '#55555d';
 
-export const SK_DARK_CONSTRAST = '#55555d';
+export const DISPLAY_STYLE = {
+    BADGE: 'badge',
```
Comment (Mario54) [inline src/js/constants/styles.js]: I wonder if we should use `Symbol()` instead of 'badge' or 'tab' here?

Comment (lemieux) [inline src/js/constants/styles.js]: why would we? it matches string values returned by the server

File: src/js/components/widget.jsx (hunk 4 above)
Comment (lemieux) [inline src/js/components/widget.jsx]: animation

File: src/js/constants/styles.js (hunk 5 above)
Comment (Mario54) [inline src/js/constants/styles.js]: oh I thought it was just constants for the widget

File: src/js/components/badge.jsx (hunk 3 above)
Comment (Mario54) [inline src/js/components/badge.jsx]: Yeah but it's exported and you said that export names should stand on their own :)

**Comment by Mario54** [comment]
👍
